http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/JsonPayloadIT.java

```diff
@@ -151,34 +145,42 @@ public void sendIncorrectContentType() throws IOException {
 
     @Test
     public void sendSpecialCharactersJSONPayload() throws Exception {
-        FootballTeam team1 = new FootballTeam(
-                "Føøbåll™ Tëâm",
-                "Rëd&Blüé",
-                3,
-                new Date(1900, 1, 1),
-                "Städiümß©",
-                Arrays.asList("لاعب Ône", "Pläÿér Twø´ñ"));
+        String jsonPayload = readJSONFile("footbal-teams_specialchars.json");
+        Response response = getApp().given()
+                .contentType(ContentType.JSON)
+                .body(jsonPayload)
+                .when()
+                .post("/football/upload-football-json")
+                .then()
+                .statusCode(CREATION_STATUS_CODE)
+                .body(containsString("Teams added successfully")).extract().response();
 
-        FootballTeam team2 = new FootballTeam(
-                "Tëâm 2",
-                "Grëèn&Yëllöw",
-                2,
-                new Date(2010, 1, 15),
-                "New Städiüm©",
-                Arrays.asList("Pläÿér プレイ", "Pläÿér Fòur"));
+        JsonPath jsonPath = response.jsonPath();
+        List<?> teams = jsonPath.getList("teams");
 
-        ObjectMapper objectMapper = new ObjectMapper();
-        String jsonPayload = objectMapper.writeValueAsString(Arrays.asList(team1, team2));
+        assertThat(teams, hasSize(2));
+        assertThat(jsonPath.getString("message"), is("Teams added successfully"));
 
+        assertThat(jsonPath.getString("teams[0].name"), containsString("Føøbåll™ Tëâm"));
+        assertThat(jsonPath.getString("teams[0].keyPlayers[0]"), is("لاعب Ône"));
+        assertThat(jsonPath.getString("teams[0].keyPlayers[1]"), is("Pläÿér Twø´ñ"));
+
+        assertThat(jsonPath.getString("teams[1].name"), containsString("Tëâm 2"));
+        assertThat(jsonPath.getString("teams[1].keyPlayers[0]"), is("Pläÿér プレイ"));
+        assertThat(jsonPath.getString("teams[1].keyPlayers[1]"), is("Pläÿér Fòur"));
+    }
+
+    @Test
+    public void sendJSONPayLoadWithNullValues() throws IOException {
+        String jsonPayload = readJSONFile("football-teams_nulls.json");
         getApp().given()
                 .contentType(ContentType.JSON)
                 .body(jsonPayload)
                 .when()
                 .post("/football/upload-football-json")
                 .then()
-                .statusCode(CREATION_STATUS_CODE)
-                .body(containsString("Teams added successfully"));
-
+                .statusCode(INTERNAL_SERVER_ERROR_STATUS_CODE)
+                .body(containsString("Unable to deserialize property 'euroCups' "));
```

Comment:
Good point because, I tried now, put not null eurocups, and keep null name team and returns `201` and the response it's something like: `{"teams":[{"colorTShirt":"White","euroCups":3}`,interesting, I will investigate it.

Reply:
It seems it's a normal behavior (https://javaee.github.io/jsonb-spec/docs/user-guide.html#null-handling)
It means that if you have a Java object with a null value for a field, JSON-B will not include that field in the resulting JSON string.
I've added another test for it.

Reply:
Oh I didn't know that. 

>By default JSON-B doesn’t serialize fields with null values.

To me it's look little strange that you are able to deserialize it if the value type is string but not when the type is integer. 

To me the correct output should be same . So to me it's look like a bug

Reply:
According docs, it's the behavior expected I think.
Look at this--> https://github.com/jakartaee/jsonb-api/issues/129

Reply:
Thanks for looking into it. I'll will take a look.